core/src/main/java/com/netease/arctic/utils/ArcticDataFiles.java

```diff
@@ -151,6 +151,6 @@ private static GenericRecord genericRecord(PartitionSpec spec) {
         return s.name();
       }
     }).collect(Collectors.toList());
-    return GenericRecord.create(spec.schema().select(collect));
+    return GenericRecord.create(spec.schema().caseInsensitiveSelect(collect));
```

Comment:
Assuming that the partition field is all lowercase is risky, as it may cause problems in systems that support uppercase fields, leading to incorrect parsing of partition data through the partition path.

Reply:
This method is defined by iceberg as
/**
   * Creates a projection schema for a subset of columns, selected by name.
   *
   * <p>Names that identify nested fields will select part or all of the field's top-level column.
   *
   * @param names String names for selected columns
   * @return a projection schema from this schema, by name
   */
In general, iceberg schema doesn't allow to see fields called 'PartitionName' and 'partitionname' at the same time, but if I don't use caseInsensitiveSelect, it will construct an incorrect GenericRecord if I query 'partitionname'  with 'PartitionName' .
So I don't think this change should be risky.